qa/standalone/osd/osd-recovery-stats.sh

```diff
@@ -190,14 +190,14 @@ function TEST_recovery_sizeup() {
     ceph osd out osd.$primary osd.$otherosd
     ceph osd pool set test size 4
     ceph osd unset norecover
-    ceph tell osd.$(get_primary $poolname obj1) debug kick_recovery_wq 0
+    # Get new primary
+    primary=$(get_primary $poolname obj1)
```

Comment:
This change works in master even without async recovery.  Looking at this now I'd say the most important thing is that after the reconfiguration the new primary needs to be selected and returned at this point.  If the primary really did change between the kick_recovery_wq and after the wait_for_clean() then we probably kicked the wrong osd.  The test will fail if the check() below uses the wrong osd's log.

Reply:
Here is some background I shared with Josh earlier on this:

- before async_recovery, the new osds in play would be [2,3,4,5] and all of them would be in the acting set. This would mean that the primary once chosen would remain the same. So the primary used to kick recovery would be the same as when we "Get new primary". In this case it would be osd 2. 

- after async recovery, the acting set becomes [4,5] in the choose_acting function with [2,3] as async recovery targets. So, osd.4 is used to kick recovery. However, when recovery completes, all the osds get back to the acting set and the primary is now osd.2. So there is a race condition, where the new primary found would be different from the primary on which the recovery was kicked. The value of the primary is then used in the check function to grep for degraded. This will fail since it greps on osd.2, while the logs exist on osd.4.

Reply:
So that mean that a recovery that use to look like [2,3,4,5]p2 now looks like [2,3]/[4,5]p4 up/acting?

Reply:
Something like this [2,4,3,5]/[4,5]

Reply:
maybe we should add async_recovery_targets and backfill_targets to the pg dout() prefix to make this clearer when peering logs aren't available